logtracer/jsonlog.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import sys
+import traceback
+from datetime import datetime
+
+from pythonjsonlogger import jsonlogger
+
+from logtracer import _b3, _global_vars
+
+LOG_SEVERITIES = {
+    'DEBUG': 'DEBUG',
+    'INFO': 'INFO',
+    'WARNING': 'WARNING',
+    'WARN': 'WARNING',
+    'ERROR': 'ERROR',
+    'EXCEPTION': 'ERROR'
+}
+
+
+class StackdriverJsonFormatter(jsonlogger.JsonFormatter):
+    def add_fields(self, log_record, record, message_dict):
+        """
+        Add GCP StackDriver fields (https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry) to JSON logger
+        output and remove default unused fields.
+        """
+
+        if record.exc_info:
+            exception_class, exception, tb = record.exc_info
+            tb_str = "\n".join(traceback.format_tb(tb))
+            record.message = f"""Exception: {exception_class.__name__}({str(exception)})\nTraceback:\n{tb_str}"""
+
+        gcp_log_record = {
+            'severity': LOG_SEVERITIES[record.levelname],
+            'time': datetime.fromtimestamp(record.created),
+            'logging.googleapis.com/sourceLocation': {
+                'file': record.pathname,
+                'line': record.lineno,
+                'function': record.funcName
+            },
+            'message': f'{record.name} - {record.message}'
+        }
+
+        b3_values = _b3.values()
```

Comment:
What does this line mean?

Reply:
From the docstring:
```
    Get the full current set of B3 values. If a span is not started then return null values by default.

    Returns:
        (dict): Contains the keys "X-B3-TraceId", "X-B3-ParentSpanId", "X-B3-SpanId", "X-B3-Sampled" and "X-B3-Flags" for the current span or subspan. NB some of the values are likely be None, but all keys will be present.
```
Do you think more could be added, or the function renamed?